### PR TITLE
Extract reflect_fields method to the repository

### PR DESCRIPTION
**Pull request by jcoyne**
This allows the repository to implement reflect_fields and allows us to
remove a conditional.

### CODE REVIEW

File: lib/blacklight/configuration/fields.rb [hunk 1]
```diff
@@ -137,17 +120,35 @@ def add_blacklight_field config_key, *args, &block
...
           return nil
         end
 
         @table[:luke_fields] ||= Rails.cache.fetch("blacklight_configuration/admin/luke", expires_in: 1.hour) do
           begin
-            if repository_class <= Blacklight::Solr::Repository
-              repository = repository_class.new(self)
-              repository.send_and_receive('admin/luke', params: { fl: '*', 'json.nl' => 'map' })['fields']
-            end
+            repository = repository_class.new(self)
```
Comment (mjgiarlo) [inline lib/blacklight/configuration/fields.rb]: Can you safely remove the condition around this?

Comment (jcoyne) [reply lib/blacklight/configuration/fields.rb]: The tests seem to pass.  I can think of no reason why we can't require repository implementations to implement this interface.

Comment (mjgiarlo) [reply lib/blacklight/configuration/fields.rb]: Looks like @cbeer last touched this code, here: https://github.com/projectblacklight/blacklight/commit/d75ff8594463f3c078ea0b0e51fc48ed15bef70e

Thoughts on that, Chris?

File: lib/blacklight/configuration/fields.rb [hunk 2]
```diff
@@ -137,17 +120,35 @@ def add_blacklight_field config_key, *args, &block
...
+          config.key = field
+          if self[config_key.pluralize][config.key]
+            self[config_key.pluralize][config.key] = config.merge(self[config_key.pluralize][config.key])
+          else
+            add_blacklight_field(config_key, config, &block)
+          end
+        end
+      end
+
       def luke_fields
```
Comment (cbeer) [inline lib/blacklight/configuration/fields.rb]: Should we rename this method?

File: lib/blacklight/configuration/fields.rb (hunk 1 above)
Comment (cbeer) [reply lib/blacklight/configuration/fields.rb]: I don't think it's a problem; I don't remember what the luke response looks like, but if we think something similar will work cross-repository, I don't see a problem with it.

**Pull request merged**